### PR TITLE
added Order model

### PR DESCRIPTION
**Pull request by DmytroFedoryshyn**

### CODE REVIEW

**Comment by DmytroFedoryshyn** [comment]
I have a question about whether it is correct to load order item's book details, including price, and then calculate total order price in OrderServiceImpl? This code is on lines 40-55 of OrderServiceImpl class. I've marked that code with question marks

File: src/main/java/bookstore/service/OrderServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+        Iterator<OrderItem> iterator = orderItems.iterator();
+
+        //?
+        while (iterator.hasNext()) {
+            OrderItem orderItem = iterator.next();
+            orderItem.setOrder(order);
+            Book book = bookRepository.getReferenceById(orderItem.getBook().getId());
+            orderItem.setBook(book);
+            orderItem.setPrice(orderItem.getBook().getPrice());
+        }
```
Comment (sokimaaa) [inline src/main/java/bookstore/service/OrderServiceImpl.java]: use for-each loop here

File: src/main/java/bookstore/controller/OrderController.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                                              + "    \"orderItems\": [\n"
+                                              + "      {\n"
+                                              + "        \"bookId\": 1,\n"
+                                              + "        \"quantity\": 2\n"
+                                              + "      }\n"
+                                              + "    ]\n"
+                                              + "}"))
+    })
+    @PostMapping
+    public OrderResponseDto placeOrder(@Valid @RequestBody OrderRequestDto order,
```
Comment (sokimaaa) [inline src/main/java/bookstore/controller/OrderController.java:60]: Only the user should be able to make an order. (Any anonymous, or admins)
```suggestion
    @PostMapping
    @PreAuthorize("hasAuthority('USER')")
    public OrderResponseDto placeOrder(@Valid @RequestBody OrderRequestDto order,
```

File: src/main/java/bookstore/controller/OrderController.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+                                                   @RequestParam(defaultValue = "10") int size,
+                                               @Parameter(description =
+                                                   "Sort order and fields",
+                                                   example = "orderDate,desc")
+                                                   @RequestParam(defaultValue = "orderDate,desc")
+                                                       String sort) {
+        Sort.Order sortOrder = sortParametersParsingUtil.parseSortOrder(sort);
+
+        Pageable pageable = PageRequest.of(page, size, Sort.by(sortOrder));
+        return orderService.getByUser((User) authentication.getPrincipal(), pageable);
```
Comment (sokimaaa) [inline src/main/java/bookstore/controller/OrderController.java]: Controller should only invoke the Service' methods.

Logic of parsing order, making pageable should be inside service layer.

File: src/main/java/bookstore/controller/BookController.java [hunk 4]
```diff
@@ -48,11 +45,7 @@ public List<BookResponseDto> getAll(
             @RequestParam(defaultValue = "10") int size,
             @Parameter(description = "Sort order and fields", example = "title,desc")
             @RequestParam(defaultValue = "title,desc") String sort) {
-
-        Sort.Order sortOrder = sortParametersParsingUtil.parseSortOrder(sort);
-
-        Pageable pageable = PageRequest.of(page, size, Sort.by(sortOrder));
-        return bookService.findAll(pageable);
+        return bookService.findAll(page, size, sort);
```
Comment (Rommelua) [inline src/main/java/bookstore/controller/BookController.java:48]: Creating Pageable in the controller was not a bad idea in my opinion)

**Review by Rommelua — APPROVED** [review]

**Pull request merged**